src/providers/postgres/qgsconnectionconfigurator.h

```diff
@@ -0,0 +1,190 @@
+/***************************************************************************
+  qgspostgresconnectionconfigurator.h  -  connection configurator class for
+  PostgeSQL based connectors.
+                             -------------------
+    begin                : 2023/06/06
+    copyright            : (C) 2023 by Alexey Karandashev
+    email                : reflectored at pm dot me
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSCONNECTIONCONFIGURATOR_H
+#define QGSCONNECTIONCONFIGURATOR_H
+
+#include <QString>
+#include <QtCore/qstring.h>
+
+#include "qgsdatasourceuri.h"
+#include "qgssettings.h"
+
+template <typename T>
+class QgsConnectionConfigurator
+{
+    static const QString connections_prefix;
+    static const QString connection_name_prefix;
+
+  public:
+    static QStringList connectionList()
+    {
+      QgsSettings settings;
+      settings.beginGroup( connections_prefix );
+      return settings.childGroups();
+    }
+
+    static QString selectedConnection()
+    {
+      QgsSettings settings;
+      return settings.value( connections_prefix + QStringLiteral( "/selected" ) ).toString();
+    }
+
+    static void setSelectedConnection( const QString &name )
+    {
+      QgsSettings settings;
+      return settings.setValue( connections_prefix + QStringLiteral( "/selected" ), name );
+    }
+
+    static QgsDataSourceUri connUri( const QString &connName )
+    {
+      QgsDebugMsgLevel( "theConnName = " + connName, 2 );
+
+      QgsSettings settings;
+
+      QString key = connections_prefix + QStringLiteral( "/%1" ).arg( connName );
+
+      QString service = settings.value( key + QLatin1String( "/service" ) ).toString();
```

Comment:
```suggestion
      const QString service = settings.value( key + QLatin1String( "/service" ) ).toString();
```

I think these can all be const (except for port).

As a general rule, if a variable can be const, mark it as const.